addons/hr_recruitment_integration_base/models/hr_job_post.py

```diff
@@ -0,0 +1,204 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from markupsafe import Markup
+
+from odoo import fields, models, _
+from odoo.exceptions import RedirectWarning
+
+
+class JobPost(models.Model):
+    _name = "hr.job.post"
+    _description = "Job Post"
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _order = "write_date desc"
+
+    job_id = fields.Many2one(
+        'hr.job', string="Job", required=True, readonly=True)
+    recruiter_id = fields.Many2one(
+        'res.users', string="Recruiter", related='job_id.user_id', readonly=True)
+    platform = fields.Many2one(
+        'recruitment.platform', string="Platform",
+        required=True, readonly=True)
+    platform_icon = fields.Binary(related="platform.avatar_128")
+    apply_method = fields.Selection([
+        ('email', 'Email'),
+    ], string='Contact Method', required=True)
+    apply_vector = fields.Char(
+        string="Contact Point",
+        help="The email address, phone number, url to send applications to.")
+    post = fields.Html(string="Post", prefetch=False, required=True)
+    status = fields.Selection([
+        ('success', 'Success'),
+        ('warning', 'Warning'),
+        ('pending', 'Pending'),
+        ('failure', 'Failure'),
+        ('expired', 'Expired'),
+        ('deleted', 'Deleted'),
+    ], string='Status', required=True, readonly=True)
+    status_message = fields.Text(
+        string="Status Message", prefetch=False, readonly=True)
+    api_data = fields.Json(string="API Data", prefetch=False, readonly=True)
+    campaign_start_date = fields.Date(
+        string="Campaign Start Date", default=fields.Date.today(),
+        help='The date when the campaign will start.')
+    campaign_end_date = fields.Date(
+        string="Campaign End Date",
+        help='The date when the campaign will end. If not set, '
+        'the campaign will run indefinitely or to the maximum allowed by a platform.')
+    company_id = fields.Many2one(related='job_id.company_id')
+
+    def _compute_display_name(self):
+        for record in self:
+            record.display_name = _(
+                '%(job)s on %(platform)s',
+                job=record.job_id.name,
+                platform=record.platform.name
+            )
+
+    def create(self, values):
+        posts = super().create(values)
+        posts._log_post_modifications()
+        return posts
+
+    def unlink(self):
+        for job_post in self:
+            job_post.sudo()._delete_post()
+        return super().unlink()
+
+    def _log_post_modifications(self, mode=None):
+        if mode is None:
+            mode = _('created')
+        for post in self:
+            post.message_post(
+                author_id=self.env.user.partner_id.id,
+                subject=_(
+                    'Job Post on %(platform)s has been modified',
+                    platform=post.platform.name
+                ),
+                body=Markup
```

Comment:
Avoid including HTML into translated terms to avoid translations issues made by unadvised non technical people.

```py
Markup(
   """<strong>%(line_1)s</strong><br/><strong>%(line_2)s</strong>""",
   line_1=_("Job Post on %(platform)s has been %(mode)s", platform=post.platform.name, mode=mode),
   line_2=…
)
```

Reply:
yup I just used tracking